src/contracts/core/PaymentCoordinator.sol

```diff
@@ -0,0 +1,501 @@
+     * their cumulativeEarnings and cumulativeClaimed. This difference is then transferred to claimerFor[claim.earner]
+     * @param claim The PaymentMerkleClaim to be processed.
+     * Contains the root index, earner, payment leaves, and required proofs
+     * @dev only callable by the valid claimer, that is
+     * if claimerFor[claim.earner] is address(0) then only the earner can claim, otherwise only
+     * claimerFor[claim.earner] can claim the payments.
+     */
+    function processClaim(
+        PaymentMerkleClaim calldata claim
+    ) external onlyWhenNotPaused(PAUSED_CLAIM_PAYMENTS) nonReentrant {
+        DistributionRoot memory root = distributionRoots[claim.rootIndex];
+        _checkClaim(claim, root);
+        // If claimerFor earner is not set, claimer is by default the earner. Else set to claimerFor
+        address earner = claim.earnerLeaf.earner;
+        address claimer = claimerFor[earner];
+        if (claimer == address(0)) {
+            claimer = earner;
+        }
+        require(msg.sender == claimer, "PaymentCoordinator.processClaim: caller is not valid claimer");
+        for (uint256 i = 0; i < claim.tokenIndices.length; i++) {
+            TokenTreeMerkleLeaf calldata tokenLeaf = claim.tokenLeaves[i];
+
+            uint256 currCumulativeClaimed = cumulativeClaimed[earner][tokenLeaf.token];
+            require(
+                tokenLeaf.cumulativeEarnings >= currCumulativeClaimed,
+                "PaymentCoordinator.processClaim: cumulativeEarnings must be gte than cumulativeClaimed"
+            );
+
+            // Calculate amount to claim and update cumulativeClaimed
+            uint256 claimAmount = tokenLeaf.cumulativeEarnings - currCumulativeClaimed;
+            cumulativeClaimed[earner][tokenLeaf.token] = tokenLeaf.cumulativeEarnings;
+
+            tokenLeaf.token.safeTransfer(claimer, claimAmount);
+            emit PaymentClaimed(root.root, earner, claimer, claim.tokenLeaves[i].token, claimAmount);
+        }
+    }
+
+    /**
+     * @notice Creates a new distribution root. activatedAt is set to block.timestamp + activationDelay
+     * @param root The merkle root of the distribution
+     * @param paymentCalculationEndTimestamp The timestamp until which payments have been calculated
+     * @dev Only callable by the paymentUpdater
+     */
+    function submitRoot(
+        bytes32 root,
+        uint32 paymentCalculationEndTimestamp
+    ) external onlyWhenNotPaused(PAUSED_SUBMIT_ROOTS) onlyPaymentUpdater {
+        require(
+            paymentCalculationEndTimestamp > currPaymentCalculationEndTimestamp,
+            "PaymentCoordinator.submitRoot: new root must be for newer calculated period"
+        );
+        require(
+            paymentCalculationEndTimestamp < block.timestamp,
+            "PaymentCoordinator.submitRoot: paymentCalculationEndTimestamp can't be in the future"
```

Comment:
Requesting removal of the `'` apostrophe in `can't`, because it's making github's syntax highlighting wonky as fuck

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/514/commits/3866a4a20ad488378275e6fdb7c5a8ac4c6c2b09